shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts

```diff
@@ -11,78 +11,63 @@ import {
 };
 
-export type DocketEntryWithWorksheet = FormattedPendingMotionDocketEntry & {
+export type FormattedPendingMotionWithWorksheet = FormattedPendingMotion & {
   docketEntryWorksheet: RawDocketEntryWorksheet;
 };
 
 export const getPendingMotionDocketEntriesForCurrentJudgeInteractor = async (
   applicationContext: IApplicationContext,
   params: { judge: string },
 ): Promise<{
-  docketEntries: DocketEntryWithWorksheet[];
+  docketEntries: FormattedPendingMotionWithWorksheet[];
 }> => {
   const { judge } = params;
   const authorizedUser = applicationContext.getCurrentUser();
   if (!isAuthorized(authorizedUser, ROLE_PERMISSIONS.JUDGE_ACTIVITY_REPORT)) {
     throw new UnauthorizedError('Unauthorized');
   }
 
-  const { results: allDocketEntries } = await applicationContext
+  const {
+    results: allPendingMotionDocketEntriesOlderThan180DaysFromElasticSearch,
+  } = await applicationContext
     .getPersistenceGateway()
     .getAllPendingMotionDocketEntriesForJudge({ applicationContext, judge });
 
   const currentDate = prepareDateFromString().toISO()!;
-
-  const batchCases = (
+  const pendingMotionDocketEntriesOlderThan180DaysFromDynamo =
     await Promise.all(
-      allDocketEntries.map(async (docketEntry: RawDocketEntry) => {
-        const fullCase: RawCase = await applicationContext
-          .getPersistenceGateway()
-          .getCaseByDocketNumber({
+      allPendingMotionDocketEntriesOlderThan180DaysFromElasticSearch.map(
+        async (docketEntry: RawDocketEntry) =>
+          await getLatestDataForPendingMotionsFromDynamo(
+            docketEntry,
             applicationContext,
-            docketNumber: docketEntry.docketNumber,
-            includeConsolidatedCases: true,
-          });
-
-        const latestDocketEntry: RawDocketEntry = fullCase.docketEntries.find(
-          de => de.docketEntryId === docketEntry.docketEntryId,
-        );
-
-        const dayDifference = calculateDifferenceInDays(
-          currentDate,
-          docketEntry.createdAt,
-        );
-
-        const updatedDocketEntry: RawDocketEntry & {
-          daysSinceCreated: number;
-          caseCaption: string;
-          consolidatedGroupCount: number;
-          leadDocketNumber?: string;
-        } = {
-          ...docketEntry,
-          ...fullCase,
-          ...latestDocketEntry,
-          consolidatedGroupCount: fullCase.consolidatedCases.length || 1,
-          daysSinceCreated: dayDifference,
-        };
-
-        return updatedDocketEntry;
-      }),
-    )
-  ).filter(de => de.pending);
-
-  const pendingMotionsDocketEntriesWithWorksheet: DocketEntryWithWorksheet[] =
-    await attachDocketEntryWorkSheets(applicationContext, batchCases);
-
-  const uniquePendingMotions: DocketEntryWithWorksheet[] =
-    removeDuplicateDocketEntries(pendingMotionsDocketEntriesWithWorksheet);
+            currentDate,
+          ),
+      ),
+    );
+
+  const removeMotionsThatHaveBeenHandledInDynamo = de => de.pending;
```

Comment:
Can this be handled in `getLatestDataForPendingMotionsFromDynamo` and only return the pending motions?